c7n/resources/ecs.py

```diff
@@ -525,3 +538,149 @@ def process_instance(self, client, cluster, instance):
+        if old_arns != 0:
+            self.log.info("Couldn't tag %d resource(s). Needs new ARN format" % old_arns)
+
+
+@ECSCluster.action_registry.register('remove-tag')
+@TaskDefinition.action_registry.register('remove-tag')
+@Service.action_registry.register('remove-tag')
+@Task.action_registry.register('remove-tag')
+@ContainerInstance.action_registry.register('remove-tag')
+class RemoveTagEcsResource(RemoveTag):
+    """Remove tag(s) from ECS resources
+    (ecs, ecs-task-definition, ecs-service, ecs-task, ecs-container-instance)
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: ecs-cluster-remove-tag
+                resource: ecs
+                filters:
+                  - "tag:BadTag": present
+                  - taggable
+                actions:
+                  - type: remove-tag
+                    tags: ["BadTag"]
+    """
+    permissions = ('ecs:UntagResource',)
+    batch_size = 1
+
+    def process_resource_set(self, client, resources, keys):
+        old_arns = 0
+        for r in resources:
+            if not ecs_taggable(self.manager.resource_type, r):
+                old_arns += 1
+                continue
+            client.untag_resource(resourceArn=r[self.id_key], tagKeys=keys)
+        if old_arns != 0:
+            self.log.info("Couldn't untag %d resource(s). Needs new ARN format" % old_arns)
+
+
+@ECSCluster.action_registry.register('mark-for-op')
+@TaskDefinition.action_registry.register('mark-for-op')
+@Service.action_registry.register('mark-for-op')
+@Task.action_registry.register('mark-for-op')
+@ContainerInstance.action_registry.register('mark-for-op')
+class MarkEcsResourceForOp(TagDelayedAction):
+    """Mark ECS resources for deferred action
+    (ecs, ecs-task-definition, ecs-service, ecs-task, ecs-container-instance)
+
+    Requires arns in new format for tasks, services, and container-instances.
+    https://docs.aws.amazon.com/AmazonECS/latest/userguide/ecs-resource-ids.html
+
+    :example:
+
+    .. code-block:: yaml
+
+        policies:
+          - name: ecs-service-invalid-tag-stop
+            resource: ecs-service
+            filters:
+              - "tag:InvalidTag": present
+              - type: taggable
+            actions:
+              - type: mark-for-op
+                op: delete
+                days: 1
+    """
+
+
+@Service.filter_registry.register('taggable')
+@Task.filter_registry.register('taggable')
+@ContainerInstance.filter_registry.register('taggable')
+class ECSTaggable(Filter):
+    """
+    Filter ECS resources to only those using long-arn format
+    https://docs.aws.amazon.com/AmazonECS/latest/userguide/ecs-resource-ids.html
+    :example:
+
+        .. code-block:: yaml
+
+            policies:
+                - name:
+                  resource: ecs-service
+                  filters:
+                    - type: taggable
+    """
+    permissions = (None,)
```

Comment:
use a real permission, or override get_permissions

Reply:
added override for get_permissions